src/main/java/com/nylas/Deltas.java

```diff
@@ -0,0 +1,232 @@
+		if(cursor == null) {
+			throw new RuntimeException("Unexpected response from the API server. Returned 200 but no 'cursor' string found.");
+		}
+
+		return cursor;
+	}
+
+	/**
+	 * Get set of Deltas since the provided cursor
+	 * <br> {@code options} is set to null.
+	 * @see Deltas#since(String, DeltaQueryOptions) 
+	 */
+	public DeltaCursor since(String cursor) throws RequestFailedException, IOException {
+		return since(cursor, null);
+	}
+
+	/**
+	 * Get set of Deltas since the provided cursor
+	 * @param cursor The cursor to query from
+	 * @param options Additional options for this Delta query
+	 * @return A {@link DeltaCursor} object containing the set of cursors since the provider cursor
+	 * @see <a href="https://developer.nylas.com/docs/api/#get/delta">Request Delta Cursors</a>
+	 */
+	public DeltaCursor since(String cursor, DeltaQueryOptions options) throws RequestFailedException, IOException {
+		HttpUrl.Builder url = deltaEndpoint().addQueryParameter("cursor", cursor);
+		if(options != null) {
+			options.toValidMap().forEach(url::addQueryParameter);
+		}
+
+		return client.executeGet(accessToken, url, DeltaCursor.class);
+	}
+
+	/**
+	 * Stream Deltas since the provided cursor
+	 * <br> {@code listener} is set to null.
+	 * <br> {@code options} is set to null.
+	 * @see Deltas#stream(String, DeltaStreamListener, DeltaQueryOptions) 
+	 */
+	public List<Delta<? extends AccountOwnedModel>> stream(String cursor) throws RequestFailedException, IOException {
+		return stream(cursor, null, null);
+	}
+
+	/**
+	 * Stream Deltas since the provided cursor
+	 * <br> {@code listener} is set to null.
+	 * @see Deltas#stream(String, DeltaStreamListener, DeltaQueryOptions)
+	 */
+	public List<Delta<? extends AccountOwnedModel>> stream(String cursor, DeltaQueryOptions options)
+			throws RequestFailedException, IOException {
+		return stream(cursor, null, options);
+	}
+
+	/**
+	 * Stream Deltas since the provided cursor
+	 * <br> {@code options} is set to null.
+	 * @see Deltas#stream(String, DeltaStreamListener, DeltaQueryOptions)
+	 */
+	public List<Delta<? extends AccountOwnedModel>> stream(String cursor, DeltaStreamListener listener)
+			throws RequestFailedException, IOException {
+		return stream(cursor, listener, null);
+	}
+
+	/**
+	 * Stream Deltas since the provided cursor
+	 * @param cursor The cursor to start streaming from
+	 * @param listener An object that implements {@link DeltaStreamListener} to listen on every Delta received
+	 * @param options Additional options for this Delta query
+	 * @return A list of all the {@link Delta} objects captured during the duration of the stream
+	 * @see <a href="https://developer.nylas.com/docs/api#get/delta/streaming">Streaming Deltas</a>
+	 */
+	public List<Delta<? extends AccountOwnedModel>> stream(String cursor, DeltaStreamListener listener, DeltaQueryOptions options)
+			throws RequestFailedException, IOException {
+		List<Delta<? extends AccountOwnedModel>> deltas = new ArrayList<>();
```

Comment:
maybe some checks for the string cursor not being null, 

the listener is checked in the while loop, but is it a valid Listener obj? 

the options is checked for null, but what if the validMap() isn't valid?

Reply:
Thanks!

1. Good call out, adding that in now.
2. If they pass in a non-null value for listener, it must be of `DeltaStreamListener` or the implementing app won't compile, right? `DeltaStreamListener` is an interface, which if implemented in a class must have a `onDelta(Delta)` method, which would make it a valid listener in this case. Correct me if I'm wrong or if I didn't get your point!
3. `toValidMap` throws an IllegalArgumentException. I'll add that note to the method javadoc.